modules/k6/k6.go

```diff
@@ -0,0 +1,89 @@
+package k6
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+// K6Container represents the K6 container type used in the module
+type K6Container struct {
+	testcontainers.Container
+}
+
+// WithTestScript mounts the given script into the ./test directory in the container
+// and passes it to k6 as the test to run.
+// The path to the script must be an absolute path
+func WithTestScript(scriptPath string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		script := filepath.Base(scriptPath)
+		target := "/home/k6x/" + script
+		req.Files = append(
+			req.Files,
+			testcontainers.ContainerFile{
+				HostFilePath:      scriptPath,
+				ContainerFilePath: target,
+				FileMode:          0o644,
+			},
+		)
+
+		// add script to the k6 run command
+		req.Cmd = append(req.Cmd, target)
+	}
+}
+
+// WithCmdOptions pass the given options to the k6 run command
+func WithCmdOptions(options ...string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Cmd = append(req.Cmd, options...)
+	}
+}
+
+// SetEnvVar adds a '--env' command-line flag to the k6 command in the container for setting an environment variable for the test script.
+func SetEnvVar(variable string, value string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Cmd = append(req.Cmd, "--env", fmt.Sprintf("%s=%s", variable, value))
+	}
+}
+
+// WithCache uses the given volume as a cache for building the k6 binary.
+// If the volume does not exists, it is created.
+func WithCache(cache string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		mount := testcontainers.ContainerMount{
+			Source: testcontainers.DockerVolumeMountSource{
+				Name: cache,
+			},
```

Comment:
@mdelapenya  I like the idea of somehow providing a way to automatically delete the cache volume, but not sure about introducing these changes in the API before carefully considering how it affects the use cases for the module.

My main problem specifically with the suggestion of suggestion of creating the volume with the name of the test session is that I'm not familiar with the `SessionID`. 

Could you please point me to more information regarding the life-cycle of this session? 

I'm particularly concerned about this comment ` to avoid collisions across multiple runs (maybe in parallel)`. Does this mean that if each test in a table-driven test runs in parallel each one will have its own cache? If so, it would be of no benefit.

Reply:
We introduced sessionID semantics here: https://github.com/testcontainers/testcontainers-go/pull/1513

Basically:

By test session/execution we mean:

- a single "go test" invocation (including flags)
- a single "go test ./..." invocation (including flags)
- the execution of a single test or a set of tests using the IDE

As a consequence, with the sole goal of aggregating test execution across multiple packages, the new code will use the parent process ID (pid) of the current process and use it to generate a unique session ID. We are using the parent pid because the current process will be a child process of:

- the process that is running the tests, e.g.: "go test";
- the process that is running the application in development mode, e.g. "go run main.go -tags dev";
- the process that is running the tests in the IDE, e.g.: "go test ./...".

At the same time, one Ryuk instance will control that test session, killing anything that is related to it: volumes, images, containers and networks.

To your question: multiple parallel() calls will share one single parent process ID, therefore they'll share the same session ID.

Reply:
Thanks for the detailed explanation.

My initial thought is that, as far as I can see  (it is quite a large PR and maybe I'm missing something) these changes are not user-facing. Therefore, making the semantics of the `WithCache` depend on a concept that the user may not be aware of, requires more documentation and can introduce some unnecessary cognitive overload.

For example
```
WithCache creates an ephemeral volume to be used as a build cache for all the tests in a test session.
This volume will be automatically deleted at the end of the session.
```

As a user, I will ask, what a session is?

Also, I think there are two clearly different scenarios for using this feature:
1. When running tests locally. I probably don't want to clean this cache ever. 
2 When running in a CI/CD. I very likely will want to clean it after each execution (a session as you describe it) to prevent volumes from leaking.

I don't have a strong opinion about which option is better: making the auto cleanup the default behavior with an option to disable it, or the other way around. But that would affect the API. 

In any case, I find changing the semantics of the API on the fly on this PR to be a little bit forced. 

I would prefer to either drop the feature until we come up with a better design or introduce it as it is (with no auto-cleanup) and implement the auto-cleanup in a follow-up PR.

Reply:
> When running tests locally. I probably don't want to clean this cache ever.
Just to clarify, this cache should not affect the test themselves as it is only used for speeding up the build process, in the same way that go has a cache for the modules. So keeping the cache across tests should not be considered a form of side-effect, like a database may be.

Reply:
Remember that with https://github.com/testcontainers/testcontainers-go/pull/1721#discussion_r1350350051 we will have clean-up for free.

About the cognitive load, I'm taking a note to add the test session semantics to our docs. Good catch!

To this point, and given the module is 100%, I'm fine with anything you consider moving to a follow-up, or to keep it at this stage. You did a good job with the code and we have a fully working module for k6! Let's improve it in follow-ups if/when needed.

Reply:
> Remember that with https://github.com/testcontainers/testcontainers-go/pull/1721#discussion_r1350350051 we will have clean-up for free.

I agree. As I mentioned before, I think that is the way to go! 

However, there are a couple of details about the API that I think need more careful consideration.

I prefer to merge this PR and open a follow-up to discuss how to handle the two scenarios described above.

Reply:
Ok, I'm resolving the comment and approving the PR, let's work on the follow-up comments then 🚀